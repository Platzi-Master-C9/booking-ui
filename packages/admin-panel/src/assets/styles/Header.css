.Header-conteiner {
  display: block;
  height: 100px;
}

.Header {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  z-index: 9;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  height: 100px;
  width: 100%;
  border-bottom: 1px solid #9b9b9b;
  padding: 0 18px 0 18px;
  background-color: #fff;
}

#Perfil {
  -webkit-box-pack: end;
      -ms-flex-pack: end;
          justify-content: end;
}

.Circulo {
  background-color: aqua;
  width: 60px;
  height: 60px;
  border-radius: 100%;
}

#Menu-hamburger {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.Company-name {
  background-color: #1572a1;
  border-radius: 50%;
  margin: 0;
  color: white;
  width: 70px;
  height: 70px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  pointer-events: none;
}

.Company-container {
  margin-left: 2em;
}

.admin {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  padding-right: 2em;
}

.Administrador {
  color: #1572a1;
  margin-right: 20px;
  font-size: 18px;
  pointer-events: none;
}

.Admin-photo {
  border-radius: 50%;
  height: 60px;
  width: 60px;
  -o-object-fit: cover;
     object-fit: cover;
}

@media only screen and (max-width: 425px) {
  .Company-name {
    width: 50px;
    height: 50px;
  }
  .Admin-photo {
    width: 40px;
    height: 40px;
  }
  .Administrador {
    font-size: 16px;
  }
}
/*# sourceMappingURL=Header.css.map */