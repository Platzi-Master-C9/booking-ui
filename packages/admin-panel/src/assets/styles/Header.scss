.Header-container {
  display: block;
  height: 100px;
}

.Header {
  display: flex;
  position: fixed;
  z-index: 9;
  justify-content: space-between;
  align-items: center;
  height: 100px;
  width: 100%;
  border-bottom: 1px solid #9b9b9b;
  padding: 0 18px 0 18px;
  background-color: #fff;
}

#Perfil {
  justify-content: end;
}

.Circulo {
  background-color: aqua;
  width: 60px;
  height: 60px;
  border-radius: 100%;
}

#Menu-hamburger {
  display: flex;
  cursor: pointer;
}

.Company-name {
  background-color: #1572a1;
  border-radius: 50%;
  margin: 0;
  color: white;
  width: 70px;
  height: 70px;
  display: flex;
  justify-content: center;
  align-items: center;
  pointer-events: none;
}

.Company-container {
  margin-left: 2em;
}

.admin {
  display: flex;
  align-items: center;
  padding-right: 2em;
}

.Administrador {
  color: #1572a1;
  margin-right: 20px;
  font-size: 18px;
  pointer-events: none;
}

.Admin-photo {
  border-radius: 50%;
  height: 60px;
  width: 60px;
  object-fit: cover;
}

@media only screen and (max-width: 425px) {
  .Company-name {
    width: 50px;
    height: 50px;
  }

  .Admin-photo {
    width: 40px;
    height: 40px;
  }

  .Administrador {
    font-size: 16px;
  }
}