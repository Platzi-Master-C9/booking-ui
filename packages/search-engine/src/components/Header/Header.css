@import url('https://fonts.googleapis.com/css2?family=Poppins&display=swap');

body{
    background: #fff;
}

.container{
    display: grid;
    grid-template-columns: 1fr 2fr 1fr;
    grid-column-gap: 30px;
    padding: 20px;
    align-items: center;
    @media (min-width: 768px) {
        padding-left: 2.5rem;
        padding-right: 2.5rem;
    };

    background: linear-gradient(180deg, #1572A1 0%, rgba(196, 196, 196, 0) 100%);
    border-radius: 30px;
}

.icon{
    width: 80px;
    height: 80px;
    
    background: #fff;
    border-style: none;
    border-radius: 100%;
    cursor: pointer;
}

.icon a{
    text-decoration: none;
}

.icon-tx{
    
    font-family: 'Poppins', sans-serif;
    font-style: normal;
    font-weight: 600;
    font-size: 36px;
    line-height: 40px;
    
    text-align: center;
    color: #1572A1;
}

.lupa {
    width: 40px;
    height: 40px;
    background-color: #1572A1;
    color: #fff;
    border-radius: 100px;
    padding: 0.5rem;
    cursor: pointer;
}  

.searchBar{
    display: grid;
    background: #fff;
    box-shadow: 0px 8px 24px rgba(0, 0, 0, 0.25);
    border-radius: 60px;
    align-items: center;
    /* font-family: 'Poppins', sans-serif;
    font-weight: 400;
    font-size: 36px; */
    
}

.wrapper{
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
    padding: 20px;
}

.linea_Caja{
    border-right: 1px solid #1572A1;
    margin-right: 5px;
}

.box{
    border-style: none;
    background: transparent;
    outline: none;
    font-family: Poppins;
    font-size: 14px;
    font-style: normal;
    font-weight: 400;
    line-height: 21px;
    letter-spacing: 0em;
    text-align: left;

}

.box_host{
    border-style: none;
    background: transparent;
    outline: none;
    font-family: Poppins;
    font-size: 14px;
    font-style: normal;
    font-weight: 400;
    line-height: 21px;
    letter-spacing: 0em;
    text-align: left;

}

.label{
    font-family: Poppins;
    font-size: 18px;
    font-style: normal;
    font-weight: 400;
    line-height: 27px;
    letter-spacing: 0em;
    text-align: left;

}

.host{
    display: flex;
    align-items: center;
    justify-content: space-evenly;
}

.icon-host{
    width: 40px;
    height: 40px;
    background-color: #1572A1;
    border-radius: 100px;
    padding: 0.5rem;
    color: #fff;
    cursor: pointer;
}

.host-link a{
    color: #fff;
    cursor: pointer;
    text-decoration: none;
    font-family: Poppins;
    font-size: 18px;
    font-style: normal;
    font-weight: 400;
    line-height: 27px;
    letter-spacing: 0em;
    text-align: left;

}

.form{
    display: flex;
}