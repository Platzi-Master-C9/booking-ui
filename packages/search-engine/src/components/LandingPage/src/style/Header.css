* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.header-container {
  width: 1200px;
  height: 100px;
  display: flex;
  justify-content: space-between;
  /* margin: 40px 60px; */
  margin: 40px;
  padding-left: 20px;
}
