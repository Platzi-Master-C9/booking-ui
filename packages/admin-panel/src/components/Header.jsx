import React from "react";

import "../assets/styles/Header.scss";
import MenuIcon from "@mui/icons-material/Menu";

export const Header = ({ showSidebar }) => (
  <div className="Header-conteiner">
    <div className="Header">
      <div className="row" id="Menu-hamburger">
        <div id="Sidebar-list-icon">
          <MenuIcon onClick={showSidebar} />
        </div>
        <div className="Company-container">
          <h1 className="Company-name"> BS</h1>
        </div>
      </div>
      <div className="admin">
        <p className="Administrador"> Administrador</p>
        <img
          src="https://i.ibb.co/TRq94N6/Foto-de-perfil-1.png"
          alt="Foto de perfil"
          border="0"
          className="Admin-photo"
        />
      </div>
      <div className="Company-container">
        <h1 className="Company-name"> BS</h1>
      </div>
      <div className="Company-container">
        <h1 className="Company-name"> BS</h1>
      </div>
    </div>
    <div className="admin">
      <p className="Administrador"> Administrador</p>
      <img
        src="https://s3-alpha-sig.figma.com/img/5854/ba20/3706ed41ce27886ff91cfe814cc60f2a?Expires=1651449600&Signature=Vn83zN3kWRivbZ564x0ADU41aoBsahjUvek9XYQeTsqr8zjBA6rOQDky1bY9~5RjzfXtbGMvCwxlepHmSEimWJmPtdrzHgMSr1DHQJlYIr7SYViNYpKsYZXfp9Ku8QVyGYTKmPER9ugXidgD~317kmfP91WBxhVQcIhrN2~VlvJXwqImilcjPkuVsVV0-gEWuaDc~20NPlEMuZCXSTmcEj1lvNENQ-TVcOCF0kqNQGlk1iiZ5Zx62fOH9llPCARv~qjVxg5Mp-VZMkyvPU2AxuaY0dREjzkWAvzNFlMtvfRgi9hqgJsN9lw7XCdCZmVj911lfCnGlyMMpM9ivrsQ0Q__&Key-Pair-Id=APKAINTVSUGEWH5XD5UA"
        className="Admin-photo"
      />
    </div>
  </div>
);
