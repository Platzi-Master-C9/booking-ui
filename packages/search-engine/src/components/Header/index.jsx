import React, { useState } from 'react';
import SearchBar from '../LandingPage/SearchBar';
import User from '../LandingPage/User';
import ModalFilter from '../Filters/ModalFilters';
import PriceFilter from '../Filters/PriceFilter';
import SizeFilter from '../Filters/SizeFilter';
import StartsFilter from '../Filters/StarsFilter';
import './Header.css';

export default function Header({ classPage, filters, handleSubmitFilters }) {
  let classHeader = (classPage !== 'landing') ? 'header__container' : `${classPage} header__container`;
  let showFilters = (!filters) ? 'hidden' : ' ';

  const [visible, setVisible] = useState(null);
  const [minPrice, setMinPrice] = useState(40000);
  const [maxPrice, setMaxPrice] = useState(4000000);
  const [stars, setStars] = useState(null);
  const [fullPlace, setFullPlace] = useState(false);
  const [privateRoom, setPrivateRoom] = useState(false);

  const toggleVisibility = (key) => !visible ? setVisible(key) : setVisible(null);

  // Size filter

  const handleMinPrice = (e) => {
    setMinPrice(e.target.value);
  }

  const handleMaxPrice = (e) => {
    setMaxPrice(e.target.value);
  }

  const resetPriceFilter = () => {
    setMinPrice(40000);
    setMaxPrice(40000000);
  }

  const submitPriceFilter = () => {
    sendFilter();
  } 

  // Size filter

  const resetSizeFilter = () => {
    setFullPlace(false);
    setPrivateRoom(false);
  }

  const submitSizeFilter = () => {
    sendFilter();
  }

  // Stars filter

  const handleStar = (numberStars) => {
    setStars(numberStars);
  }

  const resetStars = () => {
    setStars(null);
  }

  const submitStarts = () => {
    sendFilter();
  }

  // Send all filters

  const sendFilter = () => {
    handleSubmitFilters(minPrice, maxPrice, stars, fullPlace, privateRoom);
  }

  return (
    <header className={classHeader}>
      <div className="logo__header">
        <a href="/">BS</a>
      </div>
      <SearchBar />
      <User classUser={classPage} />
      <div className={`filter__container ${showFilters}`}>
        <ModalFilter 
          title='Precio'
          visible={visible === 'visiblePrice'}
          setVisible={() => toggleVisibility('visiblePrice')}
          reset={resetPriceFilter}
          submitFn={submitPriceFilter}
        >
          <PriceFilter 
            minPrice={minPrice}
            maxPrice={maxPrice}
            handleMinPrice={handleMinPrice}
            handleMaxPrice={handleMaxPrice}
          />
        </ModalFilter>

        <ModalFilter 
          title='Tipo de alojamiento'
          visible={visible === 'visibleSize'}
          setVisible={() => toggleVisibility('visibleSize')}
          reset={resetSizeFilter}
          submitFn={submitSizeFilter}
        >
          <SizeFilter 
            fullPlace={fullPlace}
            privateRoom={privateRoom}
            handleFullPlace={() => setFullPlace(!fullPlace)}
            handlePrivateRoom={() => setPrivateRoom(!privateRoom)}
          />
        </ModalFilter>

        <ModalFilter 
          title='CalificaciÃ³n'
          visible={visible === 'visibleStars'}
          setVisible={() => toggleVisibility('visibleStars')}
          reset={resetStars}
          submitFn={submitStarts}
        >
          <StartsFilter 
            stars={stars}
            handleStar={handleStar} 
          />
        </ModalFilter>
      </div>
    </header>
  );
};
