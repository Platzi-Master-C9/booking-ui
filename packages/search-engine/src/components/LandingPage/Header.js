import React from "react";
import { Logo } from "./Logo";
import { SearchBar } from "./SearchBar";
import { User } from "./User";
import "../style/Header.css";

const Header = () => {
  return (
    <header className="header-container">
      <Logo />
      <SearchBar />
      <User />
    </header>
  );
};

export { Header };
