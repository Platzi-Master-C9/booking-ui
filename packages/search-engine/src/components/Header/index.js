import React from 'react';
import { Logo } from '../LandingPage/Logo';
import { SearchBar } from '../LandingPage/SearchBar';
import { User } from '../LandingPage/User';
import './Header.css';

const Header = () => {
  return (
    <header className="header-container">
      <Logo />
      <SearchBar />
      <User />
    </header>
  );
};

export { Header };
