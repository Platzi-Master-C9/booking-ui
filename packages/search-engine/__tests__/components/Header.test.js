import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';

import Header from '../../src/components/Header';

describe('<Header />', () => {
  it('Render the header component', () => {
    const header = mount(<Header />);
    expect(header.length).toEqual(1);
  });

  it('Render the header component in the landing page', () => {
    const header = mount(<Header classPage='landing' />);

    expect(header.length).toEqual(1);
    expect(header.find('.landing').length).toEqual(2);
  });
});

describe('Header Snapshot', () => {
  it('compounding the UI of the component', () => {
    const component = create(<Header />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('compounding the UI of the component in the landing page', () => {
    const component = create(<Header classPage='landing' />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
