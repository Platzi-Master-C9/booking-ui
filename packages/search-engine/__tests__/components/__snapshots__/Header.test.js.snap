// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot compounding the UI of the component 1`] = `
<header
  className="header__container"
>
  <div
    className="logo"
  >
    <a
      href="/"
    >
      BS
    </a>
  </div>
  <div
    className="search__container"
  />
  <div
    className="users_container"
  >
    <div
      className="link host"
    >
      <a
        href="/"
      >
        Hazte anfitrión
      </a>
    </div>
    <div
      className="user"
    />
  </div>
</header>
`;

exports[`Header Snapshot compounding the UI of the component in the landing page 1`] = `
<header
  className="landing header__container"
>
  <div
    className="logo"
  >
    <a
      href="/"
    >
      BS
    </a>
  </div>
  <div
    className="search__container"
  />
  <div
    className="landing users_container"
  >
    <div
      className="link host"
    >
      <a
        href="/"
      >
        Hazte anfitrión
      </a>
    </div>
    <div
      className="user"
    />
  </div>
</header>
`;
